lab_1_keywords_tfidf/main.py

```diff
@@ -3,108 +3,164 @@
 Extract keywords based on frequency related metrics
 """
 from typing import Optional, Union
+import math
 
 
 def clean_and_tokenize(text: str) -> Optional[list[str]]:
     """
-    Removes punctuation, casts to lowercase, splits into tokens
-
-    Parameters:
-    text (str): Original text
-
-    Returns:
-    list[str]: A sequence of lowercase tokens with no punctuation
-
-    In case of corrupt input arguments, None is returned
-    """
-    pass
+       Removes punctuation, casts to lowercase, splits into tokens
+       Parameters:
+       text (str): Original text
+       Returns:
+       list[str]: A sequence of lowercase tokens with no punctuation
+       In case of corrupt input arguments, None is returned
+       """
+    if isinstance(text, str):
```

Comment:
revert the condition and return None immediately

Reply:
done